compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -1115,51 +1118,31 @@ pub fn typeid_for_instance<'tcx>(
         instance.args = strip_receiver_auto(tcx, instance.args)
     }
 
+    if let Some(impl_id) = tcx.impl_of_method(instance.def_id())
+        && let Some(trait_ref) = tcx.impl_trait_ref(impl_id)
+    {
+        // Trait methods will have a Self polymorphic parameter, where the concreteized
+        // implementatation will not. We need to walk back to the more general trait method
+        let trait_ref = trait_ref.instantiate(tcx, instance.args);
+        let invoke_ty = trait_object_ty(tcx, ty::Binder::dummy(trait_ref));
+        let method_id = tcx
+            .impl_item_implementor_ids(impl_id)
+            .items()
+            .filter_map(|(trait_method, impl_method)| {
+                (*impl_method == instance.def_id()).then_some(*trait_method)
+            })
+            .min()
```

Comment:
you dont need to do this -- just call associated_item(instance.def_id) and access the trait_item_def_id field: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/assoc/struct.AssocItem.html